cleanlab_studio/studio/enrichment.py

```diff
@@ -92,3 +107,202 @@ def to_dict(self) -> Dict[str, Any]:
             "created_at": self.created_at,
             "updated_at": self.updated_at,
         }
+
+    def preview(
+        self,
+        options: EnrichmentOptions,
+        *,
+        new_column_name: str,
+        indices: Optional[List[int]] = None,
+        disable_warnings: bool = False,
+    ) -> EnrichmentPreviewResult:
+        """Run a subset of data through the enrichment service and preview the results."""
+        extraction_pattern = None
+        replacements = list()
+
+        def _validate_tuple_is_replacement(t: Tuple) -> None:
+            if isinstance(t, tuple) and len(t) == 2 and all(isinstance(x, str) for x in t):
+                return None
+            raise ValueError(REGEX_PARAMETER_ERROR_MESSAGE)
+
+        user_input_regex = options.get("regex")
+        if user_input_regex:
+            if isinstance(user_input_regex, str):
+                extraction_pattern = user_input_regex
+            elif isinstance(user_input_regex, tuple):
+                _validate_tuple_is_replacement(user_input_regex)
+                replacements.append(
+                    {"pattern": user_input_regex[0], "replacement": user_input_regex[1]}
+                )
+            elif isinstance(user_input_regex, list):
+                for replacement in user_input_regex:
+                    _validate_tuple_is_replacement(replacement)
+                    replacements.append({"pattern": replacement[0], "replacement": replacement[1]})
+            else:
+                raise ValueError(REGEX_PARAMETER_ERROR_MESSAGE)
+
+        response = api.enrichment_preview(
+            api_key=self._api_key,
+            project_id=self._id,
+            new_column_name=new_column_name,
+            constrain_outputs=options.get("constrain_outputs"),
+            extraction_pattern=extraction_pattern,
+            indices=indices,
+            optimize_prompt=options.get("optimize_prompt"),
+            prompt=options.get("prompt"),
+            quality_preset=options.get("quality_preset"),
+            replacements=replacements,
+            tlm_options=options.get("tlm_options"),
+        )
+        epr = EnrichmentPreviewResult.from_dict(response)
+        if not disable_warnings and epr._is_timeout:
+            warnings.warn(
+                "Warning: The preview operation timed out for a subset of data. Set those results to None."
+            )
+        return epr
+
+
+class EnrichmentOptions(TypedDict):
+    """Options for enriching a dataset with a Trustworthy Language Model (TLM).
+
+    Args:
+        prompt (str): Formatted f-string, that contains both the prompt, and names of columns to embed.
```

Comment:
We're using `string.Template` for this, not f-strings! You can also link to https://docs.python.org/3/library/string.html#template-strings if you think that'll help people.

You'll also want to validate these strings locally.

Reply:
Good catch. This is copied from the hui wen original method description. 
Since you wrote that part in the backend, can you provided a meaningful docstring for `prompt`?

Reply:
[Here's what I put down earlier](https://github.com/cleanlab/cleanlab-studio-backend/blob/2e008c816cda37e8c473edc392537fa00d0df62a/backend/api/enrichment/enrichment_preview_execution_logic.py#L226). Feel free to borrow from it.